packages/lesswrong/components/posts/HideRepeatedPostsContext.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { createContext, useContext, useRef } from "react";
+
+export type HideRepeatedPostsPayload = {
+  isPostRepeated: (ownerId: string, postId: string) => boolean,
+  addPost: (ownerId: string, postId: string) => void,
+}
+
+const HideRepeatedPostsContext = createContext<HideRepeatedPostsPayload>({
+  isPostRepeated: (ownerId: string, postId: string) => false,
+  addPost: (ownerId: string, postId: string) => {},
+});
+
+export const useHideRepeatedPosts = () => {
+  const id = useRef("hrp" + Math.floor(Math.random() * 1e8));
+  const {isPostRepeated, addPost} = useContext(HideRepeatedPostsContext);
+  return {
+    isPostRepeated: isPostRepeated.bind(null, id.current),
+    addPost: addPost.bind(null, id.current),
```

Comment:
Never seen function currying in js before.

Reply:
Probably because it's super ugly!

On a related note, I'm a big fan of [Purescript](https://www.purescript.org/) which is an ML-like language that compiles to JS. To implement currying it compiles `myFunction a b c = a + b * c` into `const myFunction = (a) => (b) => (c) => a + b * c`.